apps/extension/src/ui/domains/Asset/Fiat.tsx

```diff
@@ -42,16 +44,18 @@ const DisplayValue: FC<DisplayValueProps> = ({ amount, currency, noCountUp }) =>
 
 export const Fiat = ({
   amount,
-  currency,
   className,
   noCountUp = false,
   isBalance = false,
+  hideSymbol = false,
 }: FiatProps) => {
   const { refReveal, isRevealable, isRevealed, isHidden, effectiveNoCountUp } =
     useRevealableBalance(isBalance, noCountUp)
 
   const render = amount !== null && amount !== undefined
 
+  const currency = useRecoilValue(selectedCurrencyState)
```

Comment:
Would prefer to see this wrapped in a `useCurrencySetting` hook as we have done with all other recoil state access

Reply:
Hmm, not sure whether wrapping every recoil in a hook is the best approach. Because, recoil have multiple access patterns, all with different re-render behaviour (`useRecoilState`, `useRecoilValue`, `useRecoilValueLoadable`, etc).

Though not applicable now, having direct access to recoil state allows concurrent suspense pattern like this:

```tsx
const [asyncValueA, asyncValueB, asyncValueC] = useRecoilValue(waitForAll([asyncStateA, asyncStateB, asyncStateC]));
```

Reply:
That's good to keep in mind for the future, but in this PR there's only a single access pattern used for this atom, and it's used in quite a few places, so it definitely makes sense to consolidate the behaviour to a reusable hook

Reply:
See `useToken` and `useTokenRates` for good examples of how the type of hook I think we need in this case.